src/hotspot/share/opto/superword.cpp

```diff
@@ -1822,6 +1604,11 @@ uint SuperWord::max_implemented_size(const Node_List* pack) {
   }
 }
 
+// Java API for Long.bitCount/numberOfLeadingZeros/numberOfTrailingZeros
+// returns int type, but Vector API for them returns long type. To unify
+// the implementation in backend, superword splits the vector implementation
+// for Java API into an execution node with long type plus another node
+// converting long to int.
```

Comment:
Note: copied this comment from the use-site. This one is important, and I need it inside `is_velt_basic_type_compatible_use_def`.